examples/error-handling/src/with_rejection.rs

```diff
@@ -0,0 +1,36 @@
+use axum::{extract::rejection::JsonRejection, http::StatusCode, response::IntoResponse, Json};
+use axum_extra::extract::WithRejection;
+use chrono::Utc;
+use serde_json::{json, Value};
+use thiserror::Error;
+
+pub async fn handler(
+    WithRejection(Json(value), _): WithRejection<Json<Value>, ApiError>,
+) -> impl IntoResponse {
+    Json(dbg!(value))
+}
+
+#[derive(Debug, Error)]
+pub enum ApiError {
+    #[error(transparent)]
+    JsonExtractorRejection(#[from] JsonRejection),
+}
+
+impl IntoResponse for ApiError {
+    fn into_response(self) -> axum::response::Response {
+        let payload = json!({
+            "message": self.to_string(),
+            "timestamp": Utc::now(),
+            "origin": "with_rejection"
+        });
+        let code = match self {
+            ApiError::JsonExtractorRejection(x) => match x {
+                JsonRejection::JsonDataError(_) | JsonRejection::MissingJsonContentType(_) => {
+                    StatusCode::BAD_REQUEST
+                }
+                _ => StatusCode::INTERNAL_SERVER_ERROR,
```

Comment:
I'm slightly worried that people will just copy/paste these examples so I think its important to use the right status codes.

- `JsonDataError` should be `StatusCode::UNPROCESSABLE_ENTITY`
- `JsonSyntaxError` should be `StatusCdoe::BAD_REQUEST`
- `MissingJsonContentType` should be `StatusCode::UNSUPPORTED_MEDIA_TYPE`

Same goes for the other mods.

Reply:
I copy & pasted from https://github.com/tokio-rs/axum/blob/568394a28e62904f8db4872da0d10e90d831ff42/examples/customize-extractor-error/src/main.rs#L76 so I'm afraid we should probably double check all examples too. My bad for not reading the match statement

Reply:
Ah okay yeah then that example was wrong to begin with 😜